src/components/modules/modificationsObserver.ts

```diff
@@ -10,17 +14,82 @@ export default class ModificationsObserver extends Module {
    */
   private disabled = false;
 
+  /**
+   * Blocks wrapper mutation observer instance
+   */
+  private readonly mutationObserver: MutationObserver;
+
+  /**
+   * Timeout used to batched several events in a single onChange call
+   */
+  private batchingTimeout: null | ReturnType<typeof setTimeout> = null;
+
+  /**
+   * Array of onChange events used to batch them
+   *
+   * Map is used to filter duplicated events related to the same block
+   */
+  private batchingOnChangeQueue: Map<`block:${BlockId}:event:${BlockMutationType}`, BlockMutationEvent> = new Map();
```

Comment:
I'd use an type alias for that as well. Looks like magic const imho

Reply:
Actually, much better like this:

```suggestion
  private batchingOnChangeQueue = new Map<`block:${BlockId}:event:${BlockMutationType}`, BlockMutationEvent>();
```